src/factories/kafka-rest-proxy-factory.js

```diff
@@ -533,11 +533,20 @@ angularAPP.factory('KafkaRestProxyFactory', function ($rootScope, $http, $log, $
 
     hasExtraConfig: function (topicName) {
       var extraTopicConfig = {};
+      if(isNormalTopic(topicName)) {
       angular.forEach($rootScope.topicDetails, function (detail) {
         if (detail.name === topicName) {
           extraTopicConfig = detail.configs;
         }
       });
+      } else {
+      angular.forEach($rootScope.controlTopicDetails, function (detail) {
+        if (detail.name === topicName) {
+          extraTopicConfig = detail.configs;
+        }
+      });
+      }
+
```

Comment:
how about: 
```
var topicDetails = isNormalTopic(topicName) ? $rootScope.topicDetails : $rootScope.controlTopicDetails;
angular.forEach(topicDetails, function (detail) ....
```
so there is no duplication.. 
even though it's better not to have the `$rootScope` dependency at all and pass the correct list.